libdnf/repo/Repo.cpp

```diff
@@ -1853,28 +1853,29 @@ static const char * lrLogLevelFlagToCStr(GLogLevelFlags logLevelFlag)
     return "USER";
 }
 
+static const Logger::Level lrLogLevelFlagToLevel(GLogLevelFlags logLevelFlag)
+{
+    Logger::Level level = Logger::Level::TRACE;
+    if (logLevelFlag & G_LOG_LEVEL_ERROR)
+        level = Logger::Level::ERROR;
+    else if (logLevelFlag & G_LOG_LEVEL_CRITICAL)
+        level = Logger::Level::CRITICAL;
```

Comment:
If the `G_LOG_LEVEL`s are flags (and you can have multiple levels at the same time? that's weird), shouldn't `CRITICAL` go before `ERROR`, so that if they are both set at the same time, it results in `Logger::Level::CRITICAL` here?

Also, can you add scope brackets (`{}`)to all the `if`s?

Reply:
> If the `G_LOG_LEVEL`s are flags (and you can have multiple levels at the same time? that's weird), shouldn't `CRITICAL` go before `ERROR`, so that if they are both set at the same time, it results in `Logger::Level::CRITICAL` here?

Hm, these levels actually don't correspond well to the libdnf ones. In the ``GLogLevelFlags``, ``ERROR`` is before ``CRITICAL`` as ``CRITICAL`` is probably used for critical warnings. To preserve the order, it would have to be ``Logger::Level::ERROR`` for ``G_LOG_LEVEL_CRITICAL`` and ``Logger::Level::CRITICAL`` for ``G_LOG_LEVEL_ERROR``, which is weird and as a consequence, the strings for these levels will change, but at least it will follow the ``logfilelevel`` configuration.

> Also, can you add scope brackets (`{}`)to all the `if`s?

I can add the scope brackets.

And btw, I just noticed I can also remove the ``lrLogLevelFlagToCStr`` function, since it's not used anymore.

Reply:
Hm, or probably better approach would be to have ``Logger::Level::WARNING`` for both ``G_LOG_LEVEL_WARNING`` and ``G_LOG_LEVEL_CRITICAL``, because it really is a type of warning.

Reply:
I changed it so that ``G_LOG_LEVEL_CRITICAL`` results in ``Logger::Level::WARNING``. Let me know if you have better idea.

Also, I removed the else-ifs, so now it has similar format as the previous function - should I still add the scope brackets?

Reply:
Hmm, so the `G_LOG_LEVEL_CRITICAL` is indeed documented to mean a critical warning and it also suggests the exact semantics should be defined by the application.

The g_log messages actually don't map too well to our/"standard" log levels, since `g_error` always calls terminate, so you can't log it without it being the last thing you do. So I'm thinking the better mapping would actually be `G_LOG_LEVEL_CRITICAL` -> `Logger::Level::ERROR` and `G_LOG_LEVEL_ERROR` -> `Logger::Level::CRITICAL`.

In practice it likely doesn't matter, seems we don't use either `g_error` or `g_critical` in librepo and it's perhaps a bit less intuitive. So we can keep it the way it is...

> Also, I removed the else-ifs, so now it has similar format as the previous function - should I still add the scope brackets?

I think you should add the brackets anyway, we agreed to always use them and I think it makes for cleaner code...

Reply:
I added the scope brackets.